addon/appModules/notepad++/editWindow.py

```diff
@@ -161,6 +161,21 @@ def script_reportFindResult(self, gesture):
 	script_reportFindResult.__doc__ = _("Queries the next or previous search result and speaks the selection and current line of it.")
 	script_reportFindResult.category = "Notepad++"
 
+	def script_switchDocument(self, gesture):
+		gesture.send()
+		# Get the file name from the Notepad++ title bar.
+		title = self.parent.parent.parent.windowText
```

Comment:
You aren't checking whether the title has actually changed or not, potentially causing even more confusion than intended. Please implement some failsafe mechanism for this.